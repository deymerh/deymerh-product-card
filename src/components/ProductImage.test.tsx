import React from 'react';
import renderer from 'react-test-renderer';

import { ProductCard, ProductImage } from '.';

import { product1, product2 } from '../data/products';

describe('Test in <ProductImage />', () => {
    
    test('should to show the component correctly without the image', () => {
       const wrapper = renderer.create(
           <ProductCard
            product={product1}
           >
           {
               ()=>(
                   <ProductImage img='https://deymerh.co/profile.jpg' />
               )
           }
           </ProductCard>
       )
       expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should to show the component correctly with the image', () => {
       const wrapper = renderer.create(
           <ProductCard
            product={product2}
           >
           {
               ()=>(
                   <ProductImage />
               )
           }
           </ProductCard>
       )
       expect(wrapper.toJSON()).toMatchSnapshot();
    });
    
})