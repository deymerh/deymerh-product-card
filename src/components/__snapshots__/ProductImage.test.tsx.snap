// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ProductImage /> should to show the component correctly with the image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe Mug"
    className="productImg
            undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Test in <ProductImage /> should to show the component correctly without the image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe Mug"
    className="productImg
            undefined"
    src="https://deymerh.co/profile.jpg"
  />
</div>
`;
